Tails are now organs that can be removed and reattached

Superlagg opened the pull request:
<!-- The text between the arrows are comments - they will not be visible on your PR. -->
<!-- To automatically tag this PR, add the uppercase label(s) surrounded by brackets below, for example: [LABEL] -->
[FEATURE][SPRITES]
## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->

Adds a new organ: a tail. 

Every mutantrace that previously had a tail baked into their sprite has had that tail moved into this new tail organ. 

This organ defines what to display on the (humanlike) creature that happens to own it, typically some form of tail.

Every human organholder, mutant or otherwise, gets a tail organ as well. However, for any human that normally shouldn't have a tail, this tail organ is just a tailbone, and doesn't show anything on your character. 

You can also remove these tails via surgery. Its done by doing scissor-scalpel-scissor while on disarm intent. 

As you'd expect, removing their tail removes their tail, dropping their old tail as an organ. This severed tail can be reattached to any other organholding human who, themselves, have lost their tail organ, and the tail will show up on their character.

Lizard tails preserve the coloration of their first host, persisting regardless of how many people it ends up attached to.

Here're some pics!

![tails1](https://user-images.githubusercontent.com/13302055/89247743-dd379300-d5c2-11ea-94df-e7dcacb192f3.png)
![tails2](https://user-images.githubusercontent.com/13302055/89247750-e0328380-d5c2-11ea-8aba-0d54ec393224.png)
![tails3](https://user-images.githubusercontent.com/13302055/89247755-e3c60a80-d5c2-11ea-81b2-29778ccf5896.png)

A few things I should note:
 - Skeleton tailbones can be removed with a single poke of disarm-intent scissors.
 - - No chance to fail, and they're reattached just as easily.
 - Any mutantrace that starts with a visible tail (and skeletons) become very clumsy when their tail is removed.
 - - This can be fixed by giving them another tail.
 - Just about every tail (except the tail bone) is edible and I haven't figured out how to make it stop.
 - I've added a bunch of tail macro define things in _setup.dm. Just so I don't have to remember which number means what tail.
 - - They're used by mutantraces.dm and get_desc.dm. I can move the defines to those respective files if that's a better idea.
 - Changing someone's mutantrace changes their tail to that their resulting mutant race. 
 - - It does this by deleting the previous tail and adding in a new one, which while it works, could be implemented better.
 - - Like a field that makes the proc drop the tail instead of deleting it.
 - - It also doesn't proc the organ addition or removal thing properly, so tail-loss clumsiness isnt updated through this.
 - Gotta find out how to layer the oversuit tails so they are just below the backpack layer.
 - - I tried setting it to (MOB_BACK_LAYER - 0.1), but it kept hiding under the jumpsuit.
 - Spacebux purchased mutantraces are applied directly with set_mutantrace
 - - Also added an extra arg to set_mutantrace that makes it silently delete the old tail before adding a new one
 - - This is all so spacebux tailhavers dont drop a tailbone the moment they spawn in

## Why's this needed? <!-- Describe why you think this should be added to the game. -->

I wanted to remove my lizard tail.

## Changelog
<!-- If necessary, put your changelog entry below. Otherwise, please delete it.
Use however you want to be credited in the changelog in place of CodeDude.
Use (*) for major changes and (+) for minor changes. For example: -->

```
(u)Superlagg:
(*)Tails can now be removed and reattached.
```

ZeWaka commented on _std/_setup.dm:
```diff
@@ -1150,6 +1150,17 @@ var/ZLOG_START_TIME
 #define RAILING_FASTEN 2
 
+//tail defines
+#define NO_TAIL 0
+#define MONKEY_TAIL 1
+#define LIZARD_TAIL 2
+#define COW_TAIL 3
+#define WEREWOLF_TAIL 4
+#define SKELETON_TAIL 5	// tailbone *shrug*
+#define SEAMONKEY_TAIL 6
+#define CAT_TAIL 7
+#define ROACH_TAIL 8
```
could you reverse the order, so it's like `TAIL_WEREWOLF`? Helps with pollution and organization

Superlagg replied:
Sure!

ZeWaka commented on code/datums/mutantraces.dm:
```diff
@@ -302,6 +304,49 @@
+		switch(tailnum)	// Mutantraces come with a free, new tail
+			if (NO_TAIL)	// technically you have a tail, its just invisible
+				Tf.organHolder.tail = new /obj/item/organ/tail/bone(Tf, Tf.organHolder)
+				Tf.organHolder.organ_list["tail"] = Tf.organHolder.tail
```
why are you repeating this line every time

Superlagg replied:
321? Okay, I'll put that line after the switch is complete and remove the rest.

ZeWaka commented on code/datums/mutantraces.dm:
```diff
@@ -1156,6 +1208,11 @@
+	tail = SEAMONKEY_TAIL
+
+	New()
+		..()
```
probably be better to do `. = ..()` for these

Superlagg replied:
I don't know what that does, but I'll do that once I do.

Superlagg replied:
I ended up taking out those New() procs. Turns out, I didn't need to call set_tail in there!

ZeWaka commented on code/mob/living/carbon/human.dm:
```diff
@@ -2636,6 +2642,10 @@
 				ret += organHolder.intestines
+		if (organHolder.tail)
+			processed += organHolder.tail
+			if (prob(75) && organHolder.tail.loc == src)
+				ret += organHolder.tail
```
?

Superlagg replied:
That supposedly adds the tail organ into the list of body parts that'll fly out when someone's gibbed. At least, when I set it, tails started flying out of my spaceman when they exploded.

Its set to prob(75) because that's what the other limbs were set to, and I figured that tails are _kind of_ like limbs.

ZeWaka commented on code/obj/item/organs/organ_parent.dm:
```diff
@@ -341,3 +351,12 @@
 		else
 			return 0
+
+	proc/organ_fix_colors(var/hex)
```
document what this is supposed to do please

Superlagg replied:
I'm not completely sure what it does, it was part of the lizard mutantrace datum and copied over so it'd work on organs. From what I can tell, it takes a hex color, converts it to RGB, clamps the values, then converts it back to hex.

I'll stick a comment on both of the procs explaining that.

ZeWaka commented on code/obj/item/organs/tail.dm:
```diff
@@ -0,0 +1,232 @@
+	//	var/monkey_humantail = 0	// If a monkey's getting a human tail
+
+		if ((src.type == /obj/item/organ/tail/monkey && !ismonkey(H)) || (src.type != /obj/item/organ/tail/monkey && ismonkey(H)))	// If we are trying to attach a monkey tail to a non-monkey
+			boutput(user, "That [src] doesn't fit there!")	// I'll make it work it later
```
TODO note

Superlagg replied:
Fixed.

ZeWaka commented on code/obj/item/organs/tail.dm:
```diff
@@ -0,0 +1,232 @@
+			src.donor.change_misstep_chance(10)
+			src.donor.bioHolder.AddEffect(failure_ability)
+
+/* Turns out a human tailbone'll work just fine?
+/obj/item/organ/tail/human	// some dummy tail that doesnt exist cus not everyone has a tail
+	name = "human tail"
+	desc = "Humans don't have tails... do they? They don't and you shouldn't be seeing this."
+	organ_name = "tail"
+
+	on_removal()
+		qdel(src)	// Humans dont have tails!
+*/
```
?

Superlagg replied:
Forgot to remove that. Previously, human tails and bone tails were two different things, where if a human mob didn't have a mutant race with visible tail and wasn't a skeleton, they'd get some weird dummy placeholder tail organ that'd delete itself if it ever got removed. It didn't work right, and I found out that just giving them the skeleton's tail would work just fine.

I'll remove that.

Tarmunora commented on code/datums/mutantraces.dm:
```diff
@@ -1250,6 +1307,12 @@
+	New()
+		..()
+		set_tail(mob, src.tail)
+		mob.blood_color = "#AEB555"
```
better to do this as/in a separate pr - it doesn't really relate to the rest

Superlagg replied:
Will do!

Tiggersaurus commented:
I understand why (surgery tool combo limitations) but I will be sad that Borg won't be able to help with tails :(

Tiggersaurus commented:
Also does this mean synthtails could be a thing for botany?

Superlagg commented:
> Also does this mean synthtails could be a thing for botany?

Synth and cybertails are planned, but likely wont be made until I get all this working right.

> I understand why (surgery tool combo limitations) but I will be sad that Borg won't be able to help with tails :(

I'll try to figure out a workaround, I love the idea of a borg stealing tails from non-humans.

Superlagg commented:
Humanoids and monkeys can now share tails, and skeletons can receive any tail now!

![skeltail](https://user-images.githubusercontent.com/13302055/89357517-9a82c300-d675-11ea-8967-d0ca696a45e0.png)
![monkeytail](https://user-images.githubusercontent.com/13302055/89357525-9e164a00-d675-11ea-8321-a124e095656c.png)

Superlagg commented:
Just about everything seems to work now.

I've modified the set_mutantrace proc to accept a second argument that, when set to 1, silently deletes the mutant's tail before setting their mutant race. This is so I can call it from the bank purchases and have it apply without dropping a tailbone the moment the character spawns in.

Also, bank purchased mutant races now proc set_mutantrace instead of adding a bioeffect that, itself, calls set_mutantrace. It also has the side effect of not showing the messages from the bioeffect going into bio-effect.

I've also modified the mutant race setting bioeffects to not set the mutant race if they're already that mutant race. This is mainly to prevent it from potentially causing people to drop their tail then immediately grow it back.

Kyle2143 commented:
I don't like the idea of a useless "tailbone surgery" on ordinary humans. I feel like that var should just be null and not have an extra, unused tail object.

Superlagg commented:
> I don't like the idea of a useless "tailbone surgery" on ordinary humans. I feel like that var should just be null and not have an extra, unused tail object. I felt bad enough adding each organ object to each human mob. But that's kinda how we had to do it to use the vars on each organ given our current organ code. Unless tailbones and tails are planned to have actual functions on life to them, I'm unsure here.

Part of the reason everyone has one of those useless tailbones (that I totally planned out and didn't just come up with now that you mentioned it) is so if, say, a lizard loses their tail and someone eats it, they can just harvest another one from one of the many dead humans to cure their crippling clumsiness. That, and with how easy it is to pop off a skeleton's tailbone, I didn't want to make it *too* difficult for them to find a new one. At least until I make cyber and synthtails a thing.

The other part is that I think I did have them set to null early on, and then it just stopped working after that. I forget if that actually happened though, so I'll try making humans just not have any kind of tail.

...though yeah, technically blobpeople would have a tailbone, and that's just silly.

> I think that the surgery steps miiiiight be able to be improved. We're kind of moving away from intent based surgery where we can. If we can't do this surgery independant of intents, then I think it should use grab intent because that currently is the only intent based surgery a la butt and skull removal.

Sure, I could have the surgery use grab intent. Nothing else currently uses grab-scissors as their opening move, so that could work. Any suggestions on how it could work without intents?

Superlagg commented:
Tail surgery has been changed to: Saw, Scalpel, Saw on the chest. Now cyborgs can steal tails!

Skeleton tail surgery is done with a crowbar, and now properly allows skeletons to have tails added and removed with a single step!

Superlagg commented:
Included in another PR. =3
